src/smt/theory_str_noodler/theory_str_noodler_final_check.cpp

```diff
@@ -1134,7 +1146,12 @@ namespace smt::noodler {
             // WARNING: the model generation is not supported for tag automata stuff. 
             // In order to add a support of model generation we need to handle adding axioms in the form of quantified formulae 
             // (so-far the internal solver timeouts with quntified axioms)
-            add_axiom(length_formula);
+            if(this->input_has_quantifiers) {
+                ctx.assert_expr(sat_length_formula);
+                ctx.internalize_assertions();
```

Comment:
What is this for? I would add some comment.

Reply:
You added the comment to `final_check_eh` but you should also add it here (`sat_handling()`).